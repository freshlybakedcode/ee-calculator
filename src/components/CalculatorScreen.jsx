import React from 'react';
import styled from 'styled-components';
import logo from '../logo.svg';

const Screen = styled.div`
  align-content: space-between;
  background: linear-gradient(to right, #6d6678 0%, #000000 5%);
  border-bottom: 1px solid #2c2536;
  display: flex;
  flex-direction: column;
  height: 100px;
  padding: 10px 10px 0 10px;
`;

const Branding = styled.div`
  background-color: #f9fcff;
  border-radius: 3px;
  display: flex;
  flex-direction: row;
`;

const Logo = styled.img`
  height: 25px;
  padding: 5px;
`;

const Result = styled.span`
  color: #7eedf4;
  display: flex;
  font-family: 'LCD-normal';
  font-size: 2.3em;
  justify-content: flex-end;
  letter-spacing: 4px;
  margin-top: 15px;
  text-shadow: #a9f8fd 0px 0px 2px;
`;

export const CalculatorScreen = ({ display }) => (
  <Screen>
    <Branding>
      <Logo src={logo} alt="[=]" />
    </Branding>
    <Result data-testid="calculator-screen">{display}</Result>
  </Screen>
);
