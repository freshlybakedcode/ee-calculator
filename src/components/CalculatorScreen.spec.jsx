import { render } from '@testing-library/react';
import { CalculatorScreen } from './CalculatorScreen';

describe('CalculatorScreen', () => {
  it('should render as expected and output the received props', () => {
    const { container, getByTestId } = render(
      <CalculatorScreen display="100" />
    );
    expect(container).toMatchSnapshot();
    expect(getByTestId('calculator-screen').textContent).toBe('100');
  });
});
