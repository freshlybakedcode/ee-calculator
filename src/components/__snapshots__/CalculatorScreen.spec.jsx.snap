// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculatorScreen should render as expected and output the received props 1`] = `
<div>
  <div
    class="sc-bdnxRM iuwJzS"
  >
    <div
      class="sc-gtsrHT lgUMTJ"
    >
      <img
        alt="[=]"
        class="sc-dlnjwi jTFNGQ"
        src="logo.svg"
      />
    </div>
    <span
      class="sc-hKFxyN gLFFat"
      data-testid="calculator-screen"
    >
      100
    </span>
  </div>
</div>
`;
